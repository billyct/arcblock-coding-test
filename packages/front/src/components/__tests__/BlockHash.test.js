import {render} from '@testing-library/react'

jest.mock('../../hooks/useBlock')
const useBlock = require('../../hooks/useBlock')

jest.mock('../CopyButton', () => 'copy-button-component')
const BlockHash = require('../BlockHash').default

afterAll(() => {
  jest.resetModules()
})

describe.each([
  [
    {
      block: {hash: 'a3e2bcc9a5f776112497a32b05f4b9e5b2405ed9'},
    }
  ],
  [
    {block: undefined}
  ]
])('test BlockHash Component with useBlock(%o)', data => {
  beforeAll(() => {
    useBlock.__setMockData(data)
  })

  it('should render correct', function () {
    const {container} = render(<BlockHash/>)
    expect(container).toMatchSnapshot()
  })

  if (data.block) {
    test(`BlockHash__span should have text: ${data.block.hash}`, function () {
      const {getByTestId} = render(<BlockHash/>)
      expect(getByTestId('BlockHash__span')).toHaveTextContent(data.block.hash)
    })
  }
})





