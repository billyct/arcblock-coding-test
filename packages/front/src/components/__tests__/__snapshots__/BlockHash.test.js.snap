// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test BlockHash Component with useBlock({ block: { hash: 'a3e2bcc9a5f776112497a32b05f4b9e5b2405ed9' } }) should render correct 1`] = `
<div>
  <div
    class="rounded shadow  py-6 bg-white mb-4"
  >
    <h3
      class="text-xl mb-4 px-6"
    >
      Block Hash
    </h3>
    <div
      class="px-6"
    >
      <div
        class="flex items-center"
      >
        <span
          class="text-sm mr-2 break-all"
          data-testid="BlockHash__span"
        >
          a3e2bcc9a5f776112497a32b05f4b9e5b2405ed9
        </span>
        <copy-button-component
          text="a3e2bcc9a5f776112497a32b05f4b9e5b2405ed9"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`test BlockHash Component with useBlock({ block: undefined }) should render correct 1`] = `<div />`;
